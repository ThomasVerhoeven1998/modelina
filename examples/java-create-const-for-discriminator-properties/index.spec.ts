const spy = jest.spyOn(global.console, 'log').mockImplementation(() => {
  return;
});
import { generate } from './index';

describe('Should be able to generate const for discriminator properties', () => {
  afterAll(() => {
    jest.restoreAllMocks();
  });
  test('and should log expected output to console', async () => {
    await generate();
    expect(spy.mock.calls.length).toEqual(5);
    expect(spy.mock.calls.map((call) => call)).toMatchSnapshot();
  });
});
