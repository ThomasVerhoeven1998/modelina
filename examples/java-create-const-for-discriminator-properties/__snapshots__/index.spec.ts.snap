// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be able to generate const for discriminator properties and should log expected output to console 1`] = `
Array [
  Array [
    "/**
 * Pet represents a union of types: Fish, Bird, FlyingFish
 */
public interface Pet {
  String getPetType();
}",
  ],
  Array [
    "public class Owner {
  private String name;
  private Pet[] pets;
  private Map<String, Object> additionalProperties;

  public String getName() { return this.name; }
  public void setName(String name) { this.name = name; }

  public Pet[] getPets() { return this.pets; }
  public void setPets(Pet[] pets) { this.pets = pets; }

  public Map<String, Object> getAdditionalProperties() { return this.additionalProperties; }
  public void setAdditionalProperties(Map<String, Object> additionalProperties) { this.additionalProperties = additionalProperties; }
}",
  ],
  Array [
    "public class Fish implements Pet {
  private final String petType = \\"Fish\\";
  private Map<String, Object> additionalProperties;

  public String getPetType() { return this.petType; }

  public Map<String, Object> getAdditionalProperties() { return this.additionalProperties; }
  public void setAdditionalProperties(Map<String, Object> additionalProperties) { this.additionalProperties = additionalProperties; }
}",
  ],
  Array [
    "public class Bird implements Pet {
  private final String petType = \\"Bird\\";
  private String breed;
  private Map<String, Object> additionalProperties;

  public String getPetType() { return this.petType; }

  public String getBreed() { return this.breed; }
  public void setBreed(String breed) { this.breed = breed; }

  public Map<String, Object> getAdditionalProperties() { return this.additionalProperties; }
  public void setAdditionalProperties(Map<String, Object> additionalProperties) { this.additionalProperties = additionalProperties; }
}",
  ],
  Array [
    "public class FlyingFish implements Pet, Fish {
  private final String petType = \\"FlyingFish\\";
  private Boolean hasWings;
  private Map<String, Object> additionalProperties;

  public String getPetType() { return this.petType; }

  public Boolean getHasWings() { return this.hasWings; }
  public void setHasWings(Boolean hasWings) { this.hasWings = hasWings; }

  public Map<String, Object> getAdditionalProperties() { return this.additionalProperties; }
  public void setAdditionalProperties(Map<String, Object> additionalProperties) { this.additionalProperties = additionalProperties; }
}",
  ],
]
`;
